main.nf

```diff
@@ -548,7 +549,7 @@ process Prepare_for_Topup {
     set sid, "${sid}_${rev}b0_mean.nii.gz", val(rev) into simple_b0_for_topup
 
   when:
-    params.run_topup && params.run_eddy
+    params.run_topup && params.run_eddy && number_rev_dwi == 0
```

Comment:
Why do you skip this step when there is reverse dwis ? This still needs to run to extract b0 volumes for Topup. The way it is implemented, it should extract them on both volumes, to be then ordered correctly by the operations on line 561 to 575.

Reply:
I need to find a way to skip this when there is already a b0 or rev_b0. Don't know yet how to get there.

Reply:
ok first try:
```
if not b0:
- run prepare_for_topup on dwi
if not rev_b0 and rev_dwi:
- run prepare_for_topup on rev_dwi
```